File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        int n = s.size();
        for (int i = 0; i < n / 2; i++) {
            s[i] = (char) s[i] + s[n - i - 1];
            s[n - i - 1] = (char) s[i] - s[n - i - 1];
            s[i] = (char) s[i] - s[n - i - 1];
        }
    }
};


// class Solution {
// public:
//     void reverseString(vector<char>& s) {
//         int n = s.size();
//         for (int i = 0; i < n / 2; i++) {
//             s[i] = (char) s[i] ^ s[n - i - 1];
//             s[n-i-1] = (char) s[i] ^ s[n - i - 1];
//             s[i] = (char) s[i] ^ s[n - i - 1];
//         }
//     }
// };


// class Solution {
// public:
//     void reverseString(vector<char>& s) {
//         int n = s.size();
//         for (int i = 0; i < n / 2; i++) {
//             swap(s[i], s[n-i-1]);
//         }
//     }
// };

File: LeetCode/0344-reverse-string/0344-reverse-string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        n = len(s)
        for i in range(n // 2):
            s[i] = chr(ord(s[i]) ^ ord(s[n - i - 1]))
            s[n - i - 1] = chr(ord(s[i]) ^ ord(s[n - i - 1]))
            s[i] = chr(ord(s[i]) ^ ord(s[n - i - 1]))



# class Solution:
#     def reverseString(self, s: List[str]) -> None:
#         """
#         Do not return anything, modify s in-place instead.
#         """
#         n = len(s)
#         for i in range(n // 2):
#             s[i] = chr(ord(s[i]) + ord(s[n - i - 1]))
#             s[n - i - 1] = chr(ord(s[i]) - ord(s[n - i - 1]))
#             s[i] = chr(ord(s[i]) - ord(s[n - i - 1]))
 

# class Solution:
#     def reverseString(self, s: List[str]) -> None:
#         """
#         Do not return anything, modify s in-place instead.
#         """
#         n = len(s)
#         for i in range(n // 2):
#             s[i] , s[n - i - 1] = s[n - i - 1], s[i]